apollo-runtime/src/main/java/com/apollographql/apollo/cache/normalized/lru/LruNormalizedCache.java

```diff
@@ -14,17 +16,27 @@
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
+/**
+ * A {@link NormalizedCache} backed by an in memory {@link Cache}. Can be configured with an optional secondaryCache
+ * {@link NormalizedCache}, which will be used as a backup if a {@link Record} is not present in the primary cache.
+ *
+ * A common configuration is to have secondary {@link com.apollographql.apollo.cache.normalized.sql.SqlNormalizedCache}
+ * cache.
+ */
 public final class LruNormalizedCache extends NormalizedCache {
 
   private final Cache<String, Record> lruCache;
-  private final Optional<NormalizedCache> secondaryCacheStore;
+  private final Optional<NormalizedCache> secondaryCache;
 
-  public LruNormalizedCache(EvictionPolicy evictionPolicy) {
-    this(evictionPolicy, null);
-  }
-
-  public LruNormalizedCache(EvictionPolicy evictionPolicy, NormalizedCache secondaryNormalizedCache) {
-    this.secondaryCacheStore = Optional.fromNullable(secondaryNormalizedCache);
+  LruNormalizedCache(RecordFieldAdapter recordFieldAdapter,
+      EvictionPolicy evictionPolicy,
+      Optional<NormalizedCacheFactory> secondaryNormalizedCache) {
+    super(recordFieldAdapter);
+    if (secondaryNormalizedCache.isPresent()) {
```

Comment:
nit: I guess it would be better:
```
this.secondaryCache = secondaryNormalizedCache.transform(it -> it.createNormalizedCache(recordFieldAdapter)));
```